om4labs/diags/generic_yz_annual_bias_1x1deg/generic_yz_annual_bias_1x1deg.py

```diff
@@ -0,0 +1,385 @@
+    elif "depth" in list(ds_static.variables):
+        depth = ds_static.depth.to_masked_array()
+    else:
+        raise ValueError("Unable to find depth field.")
+    depth = np.where(np.isnan(depth), 0.0, depth)
+    depth = depth * -1.0
+
+    dsmodel = xr.open_mfdataset(
+        dictArgs["infile"], combine="by_coords", decode_times=False
+    )
+
+    if dictArgs["obsfile"] is not None:
+        # priority to user-provided obs file
+        dsobs = xr.open_mfdataset(
+            dictArgs["obsfile"], combine="by_coords", decode_times=False
+        )
+    else:
+        # use dataset from catalog, either from command line or default
+        cat_platform = "catalogs/obs_catalog_" + dictArgs["platform"] + ".yml"
+        catfile = pkgr.resource_filename("om4labs", cat_platform)
+        cat = intake.open_catalog(catfile)
+        dsobs = cat[dictArgs["dataset"]].to_dask()
+
+    # read in model and obs data
+    datamodel = read_data(dsmodel, dictArgs["possible_variable_names"])
+    dataobs = read_data(dsobs, dictArgs["possible_variable_names"])
+
+    # reduce data along time, here mandatory
+    if ("assigned_time" in datamodel.dims) and (len(datamodel["assigned_time"]) > 1):
+        warnings.warn(
+            "input dataset has more than one time record, "
+            + "performing non-weighted average"
+        )
+        datamodel = simple_average(datamodel, "assigned_time")
+    if ("assigned_time" in dataobs.dims) and len(dataobs["assigned_time"]) > 1:
+        warnings.warn(
+            "reference dataset has more than one time record, "
+            + "performing non-weighted average"
+        )
+        dataobs = simple_average(dataobs, "assigned_time")
+
+    datamodel = datamodel.squeeze()
+    dataobs = dataobs.squeeze()
+
+    # check final data is 3d
+    assert len(datamodel.dims) == 3
+    assert len(dataobs.dims) == 3
+
+    # check consistency of coordinates
+    assert np.allclose(datamodel["assigned_lon"], dataobs["assigned_lon"])
+    assert np.allclose(datamodel["assigned_lat"], dataobs["assigned_lat"])
+
+    # homogeneize coords
+    dataobs = copy_coordinates(datamodel, dataobs, ["assigned_lon", "assigned_lat"])
+
+    # restrict model to where obs exists
+    datamodel = datamodel.where(dataobs)
+
+    # dump values
+    model = datamodel.to_masked_array()
+    obs = dataobs.to_masked_array()
+    x = datamodel["assigned_lon"].values
+    y = datamodel["assigned_lat"].values
+    z = datamodel["assigned_depth"].values
+
+    # convert z to negative values
+    z = z * -1
+
+    # compute area
+    if "areacello" in dsmodel.variables:
+        area = dsmodel["areacello"].values
+    else:
+        if (model.shape[-2], model.shape[-1]) == (180, 360):
+            area = compute_area_regular_grid(dsmodel)
+        else:
+            raise IOError("no cell area provided")
+
+    return y, z, depth, area, codes, model, obs
+
+
+def parse(cliargs=None, template=False):
```

Comment:
should we use the new default parser here?

Reply:
This PR has been hanging around for a few weeks so I'd rather close the book on this first, then update all of the parsers at one time on a new branch.

Reply:
works for me